packages/nx/src/command-line/explain.ts

```diff
@@ -0,0 +1,94 @@
+import * as chalk from 'chalk';
+import { join, relative } from 'path';
+import { nxConfigExplainer } from '../explain/config-explainers/nx-json';
+import { workspaceRoot } from '../utils/app-root';
+import { readJsonFile } from '../utils/fileutils';
+import { output } from '../utils/output';
+
+export interface ExplainArgs {
+  config: string;
+}
+
+export const defaultNxJsonPath = join(workspaceRoot, 'nx.json');
+
+export function explainHandler(args: ExplainArgs) {
+  if (args.config === 'nx.json') {
+    args.config = defaultNxJsonPath;
+  }
+  const resolvedPath = require.resolve(args.config, {
+    paths: [process.cwd()],
+  });
+  const nxJson = readJsonFile(resolvedPath);
+
+  output.log({
+    title:
+      resolvedPath === defaultNxJsonPath
+        ? 'Explanation of your current nx.json settings'
+        : `Explanation of the nx.json settings in ${relative(
+            process.cwd(),
+            resolvedPath
+          )}`,
+  });
+
+  Object.entries(nxJson).forEach(([key, value], i, arr) => {
+    const explainer = nxConfigExplainer[key];
+
+    if (i === 0) {
+      console.log('\n');
+    }
+    const linesToLog = [];
```

Comment:
Can we rewrite this code to be cleaner?
```
stripIndents`
multi
line
string
`
```

Reply:
The indentation is significant within this output so `stripIndents` is not desirable, and there are conditionals and loops (which would be harder to read if nested within the structure of a template), so I would push back on this suggestion